src/checker.rs

```diff
@@ -690,6 +697,128 @@ pub(crate) enum CheckerInst {
     },
 }
 
+#[derive(Debug)]
+struct DebugLocationEntry {
+    vreg: VReg,
+    alloc: Allocation,
+    label: u32,
+}
+
+#[derive(Debug)]
+struct DebugLocations {
+    expected_vreg_locations: FxHashMap<(VReg, (ProgPoint, ProgPoint), u32), Allocation>,
+}
+
+impl DebugLocations {
+    fn ranges_overlaps(
+        (start_point, end_point): (ProgPoint, ProgPoint),
+        (start_inst, end_inst): (Inst, Inst),
+    ) -> Option<(ProgPoint, ProgPoint)> {
+        if end_inst <= start_point.inst() || start_inst >= end_point.inst() {
+            None
+        } else {
+            let point0 = if start_point.inst() >= start_inst {
+                start_point
+            } else {
+                ProgPoint::before(start_inst)
+            };
+            let point1 = if end_point.inst() < end_inst {
+                end_point
+            } else {
+                ProgPoint::before(end_inst)
+            };
+            Some((point0, point1))
+        }
+    }
+
+    fn new<F: Function>(f: &F, output: &Output) -> Self {
+        let mut expected_vreg_locations = FxHashMap::default();
+        for (label, start_point, end_point, alloc) in &output.debug_locations {
+            for (vreg, start_inst, end_inst, in_label) in f.debug_value_labels() {
```

Comment:
I'm a bit worried that this could result in quadratic runtime during fuzzing (and that the fuzzer will discover and exercise this worst case) -- could we optimize this somehow by building a map? For example we could have a label -> [array of allocs] dense-map, where [array of allocs] is indexed by the ProgPoint's index; then when processing `debug_value_labels` we can cross-reference that.